### PR TITLE
Add binary integration tests for CLI commands

### LINKED ISSUE #5: Add binary integration tests for CLI commands
## Context

All 56 existing tests are unit and integration tests inside `src/`. There are no binary-level integration tests that exercise the actual compiled `dispatch` binary end-to-end. No `tests/` directory exists.

The gap analysis noted `assert_cmd` as an unused dev-dependency, but it has since been removed from `Cargo.toml`. The CLAUDE.md testing guidelines recommend `assert_cmd` for binary integration tests.

Binary integration tests would catch issues that unit tests can't — argument parsing edge cases, exit code correctness, stdout/stderr separation, and end-to-end broker communication through the actual CLI interface.

### 1. Add `assert_cmd` dev-dependency

```bash
cargo add assert_cmd --dev
```

### 2. Create `tests/cli.rs` with binary integration tests

Tests should exercise the compiled `dispatch` binary using `assert_cmd::Command`. Each test should start a broker, run commands against it, and verify the output.

### Suggested test cases

| Test | What it verifies |
|------|-----------------|
| `dispatch --help` exits 0 and prints usage | Basic binary works, help text renders |
| `dispatch` with no args exits non-zero | `arg_required_else_help` is working |
| `dispatch serve` starts and accepts connections | Broker actually binds the socket |
| `dispatch register` returns JSON with `worker_id` | Registration works end-to-end through CLI |
| `dispatch team` returns valid JSON with worker list | Team listing works, output is machine-readable |
| `dispatch send` to a valid worker returns success | Message delivery works through CLI |
| `dispatch send` to an invalid worker returns error | Error handling works end-to-end |
| `dispatch listen` with timeout returns timeout response | Long-poll timeout works through CLI |
| `dispatch heartbeat` with valid worker returns success | Heartbeat works through CLI |
| stdout is clean JSON, status messages go to stderr | Output separation is correct |

### Key considerations

- Each test needs its own broker instance on a unique socket to avoid conflicts. Use `tempfile::TempDir` for the dispatch state directory and set a unique `--cell-id` per test.
- Tests that need a running broker should start `dispatch serve` as a background process, wait for the socket to appear, run commands, then clean up.
- Use `assert_cmd`'s `Assert` methods to check exit codes, stdout content (valid JSON), and stderr content.
- Tests run in parallel by default in Rust — each test must be fully isolated (own temp dir, own cell ID, own broker process).
- The broker uses Unix domain sockets, so these tests will only run on Unix platforms. Add `#[cfg(unix)]` if needed, or note the platform requirement.

## Acceptance criteria

- [ ] `assert_cmd` is in `[dev-dependencies]` in `Cargo.toml`
- [ ] `tests/cli.rs` exists with at least 5 binary integration tests
- [ ] Tests exercise the actual compiled `dispatch` binary, not library functions
- [ ] Each test is isolated (own temp dir, own broker instance, own cell ID)
- [ ] Tests verify exit codes, JSON validity of stdout, and correct stderr behavior
- [ ] All tests pass with `cargo test`
- [ ] Tests don't conflict with the existing 56 unit/integration tests in `src/`

## References

- Gap 3 in `tasks/dispatch_v1_gap_analysis.md`
- CLAUDE.md "Testing" section — recommends `assert_cmd` for binary integration tests
- No `tests/` directory currently exists

### PR DESCRIPTION
**Pull request by codesoda**
## Summary
- Adds `assert_cmd` and `predicates` as dev-dependencies
- Creates `tests/cli.rs` with 10 binary integration tests exercising the compiled `dispatch` binary end-to-end
- Each test is fully isolated: own `TempDir`, own broker process, own `--cell-id`

## Tests added

| Test | What it verifies |
|------|-----------------|
| `help_exits_zero_and_shows_usage` | `--help` exits 0, prints usage text |
| `no_args_exits_non_zero` | No args exits non-zero (`arg_required_else_help`) |
| `serve_creates_socket_file` | Broker creates `.dispatch/<cell_id>.sock` |
| `register_returns_worker_id` | Registration returns JSON with `worker_id` |
| `team_lists_registered_workers` | Team lists registered workers with correct fields |
| `send_and_listen_round_trip` | Send + listen delivers message end-to-end |
| `send_to_invalid_worker_returns_error_response` | Error JSON for missing recipient |
| `heartbeat_renews_worker_ttl` | Heartbeat returns updated `expires_at` |
| `listen_times_out_with_no_messages` | Listen timeout returns timeout response |
| `stdout_is_json_stderr_is_empty_on_success` | stdout is clean JSON, stderr empty on success |

## Test plan
- [x] `cargo fmt --check` passes
- [x] `cargo clippy --all-targets -- -D warnings` passes
- [x] `cargo test` passes (66 total: 56 existing + 10 new)
- [x] New tests don't conflict with existing unit/integration tests

Closes #5

### CODE REVIEW

File: tests/cli.rs [hunk 1]
```diff
@@ -0,0 +1,391 @@
+use std::process::{Child, Command};
```
[the remark this answers is not in the document]
Comment (codesoda) [reply tests/cli.rs:1]: Skipping the `#![cfg(unix)]` gate — this is a test file, not library code, so a compile error on Windows is clear enough. The broker itself uses Unix domain sockets so these tests can't run on Windows regardless.

File: tests/cli.rs [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+    let listen_out = dispatch_cmd(&dir, cell_id)
+        .args(["listen", "--worker-id", &worker_id, "--timeout", "1"])
+        .assert()
+        .success()
+        .get_output()
+        .stdout
+        .clone();
+
+    let json: serde_json::Value = serde_json::from_slice(&listen_out).unwrap();
+    assert_eq!(json["status"], "ok");
```
[the remark this answers is not in the document]
Comment (codesoda) [reply tests/cli.rs:356]: Addressed in 07102ad — the timeout test now also asserts that `body`, `message_id`, and `expires_at` are absent, confirming it's a timeout response and not a message delivery or heartbeat ack.
